pkg/controller/elasticsearch/driver/driver.go

```diff
@@ -115,6 +115,10 @@ var _ commondriver.Interface = &defaultDriver{}
 func (d *defaultDriver) Reconcile(ctx context.Context) *reconciler.Results {
 	results := reconciler.NewResult(ctx)
 
+	// reset the Elasticsearch health to 'unknown' so that if reconciliation fails before the observer has had a chance to get it,
+	// we stop reporting a state that may be out of date
+	d.ReconcileState.UpdateElasticsearchPendingUnknownHealth()
```

Comment:
Would it not be easier to change the `NewState` constructor to initialise the state with `Unknown`?

Reply:
Much better indeed! And this way, we keep the other information (number of nodes, version, ...) that are retrieved from the existing custom resource, therefore still valid.